pydicom/config.py

```diff
@@ -236,6 +240,13 @@ def writing_validation_mode(self, value: int) -> None:
 """
 
 
+def reset_data_element_callback():
+    """Reset both :class:`Settings` and legacy ``data_element_callback``."""
+    settings.reset_data_element_callback()
+    data_element_callback = None
```

Comment:
I'm not sure the old reset function call should also mix in changing the new settings module.  If the new system is used at all, the old one should be ignored.